esp/bindings/SOAP/Platform/soapbind.cpp

```diff
@@ -87,30 +87,19 @@ const char * CHttpSoapBinding::getTransportType()
     return "http";
 }
 
-static CSoapFault* makeSoapFault(CHttpRequest* request, IMultiException* me, const char *ns)
+static CSoapFault* makeSoapFault(CHttpRequest* request, IMultiException* me, const char *ns, const char* schemaLocation)
 {
-    const char* svcName = request->queryServiceName();
-    if (svcName && *svcName)
+    if (!isEmptyString(schemaLocation))
     {
-        const char* method = request->queryServiceMethod();
-        StringBuffer host;
-        const char* wsdlAddr = request->queryParameters()->queryProp("__wsdl_address");
-        if (wsdlAddr && *wsdlAddr)
-            host.append(wsdlAddr);
-        else
-        {
-            host.append(request->queryHost());
-            if (request->getPort()>0)
-              host.append(":").append(request->getPort());
-        }
-        
-        VStringBuffer ns_ext("xmlns=\"%s\""
-            " xsi:schemaLocation=\"%s %s/%s/%s?xsd\"", 
-            ns, ns, host.str(), svcName, method ? method : "");
+        StringBuffer ns_ext;
+        appendXMLAttr(ns_ext, "xmlns", ns, nullptr, true);
+        appendXMLAttr(ns_ext, "schemaLocation", schemaLocation, "xsi", true);
         return new CSoapFault(me, ns_ext);
     }
-
-    return new CSoapFault(me);
+    else
```

Comment:
You don't need to add the else here because the "if" above returns from the function.